common/fixation_entry.go

```diff
@@ -252,10 +252,14 @@ func (fs *FixationStore) AppendEntry(
 	latestEntry, found := fs.getUnmarshaledEntryForBlock(ctx, safeIndex, block)
 
 	var deleteAt uint64 = math.MaxUint64
+	var latest bool
 
 	// if latest entry is not found, this is a first version entry
 	if !found {
 		fs.setEntryIndex(ctx, safeIndex, true)
+		if block <= ctxBlock {
```

Comment:
this if is happening 3 times in this code, here, at 281 and 320.
can just set latest = block <= ctxBlock at the beginning and use it

Reply:
I'm unsure it improves readability or performance in a noticeable way. Anyway, the test is not for "latest" but rather for "(not) future entry". I'll consider this change as I work on allow append-after-delete (CNS-496).